python/mage/node_classification/utils/extract_from_database.py

```diff
@@ -0,0 +1,262 @@
+import random
+from collections import Counter
+import torch_geometric.transforms as T
+import typing
+
+def extract_from_database(
+    ctx: mgp.ProcCtx,
+    train_ratio: float,
+    features_name: str,
+    class_name: str,
+) -> typing.Tuple[HeteroData, str, typing.Dict, typing.Dict]:
+    
+    data = HeteroData()
+    
+
+    #################
+    # NODES
+    #################
+    nodes = list(iter(ctx.graph.vertices))
+    node_types = []
+    embedding_lengths = {}
+    observed_attribute = None
+    
+    for i in range(len(nodes)):
+        if type(nodes[i].properties.get(features_name)) == type(None):
+            continue
+        node_types.append(nodes[i].labels[0].name)
+        if nodes[i].labels[0].name not in embedding_lengths:
+            embedding_lengths[nodes[i].labels[0].name] = len(
+                nodes[i].properties.get(features_name)
+            )
+
+        # find attribute which has classification labels
+        if observed_attribute == None and type(
+            nodes[i].properties.get(class_name)
+        ) != type(None):
+            observed_attribute = nodes[i].labels[0].name
+
+    if(node_types == []):
+        raise Exception("no features vectors found")
+        
+    node_types = Counter(node_types)
+
+    append_counter, reindexing, inv_reindexing = {}, {}, {}
+
+    for k, v in node_types.items():
+        data[k].x = torch.tensor(
+            np.zeros((v, embedding_lengths[k])), dtype=torch.float32
+        )
+        if k == observed_attribute:
+            data[k].y = torch.tensor(np.zeros((v,), dtype=int), dtype=torch.long)
+            data[k].train_mask = torch.tensor(
+                np.zeros((v,), dtype=int), dtype=torch.bool
+            )
+            data[k].val_mask = torch.tensor(np.zeros((v,), dtype=int), dtype=torch.bool)
+            masks = torch.tensor(np.zeros((v,), dtype=int), dtype=torch.bool)
+
+        append_counter[k] = 0
+        reindexing[k] = {}
+        inv_reindexing[k] = {}
+
+    for i in range(len(nodes)):
+        if type(nodes[i].properties.get(features_name)) == type(None):
+            continue
+
+        t = nodes[i].labels[0].name
+
+        data[t].x[append_counter[t]] = np.add(
+            data[t].x[append_counter[t]],
+            np.array(nodes[i].properties.get(features_name)),
+        )
+        reindexing[t][append_counter[t]] = nodes[i].id
+        inv_reindexing[t][nodes[i].id] = append_counter[t]
+
+        if t == observed_attribute:
+            data[t].y[append_counter[t]] = (int)(nodes[i].properties.get(class_name))
+
+        append_counter[t] += 1
+
+    print(node_types)
+
+    #################
+    # EDGES
+    #################
+    edges = []
+    edge_types = []
+    append_counter = {}
+
+    for vertex in ctx.graph.vertices:
+        for edge in vertex.out_edges:
+            f, t, o = (
+                edge.from_vertex.labels[0].name,
+                edge.type.name,
+                edge.to_vertex.labels[0].name,
+            )
```

Comment:
Use verbose names
Avoid using `f`, `t`, `o`, and similar namings because whoever reads the code needs to guess what this really is.